src/components/template/TemplateEditor.tsx

```diff
@@ -171,9 +171,29 @@ export default function TemplateEditor() {
           },
         });
       } else if (isReorderingInsideTopics) {
+        // TODO: Put this into a function because it is repeated ------
+        const topicId = dragDestination?.droppableId
+          ? +dragDestination.droppableId
+          : undefined;
+        const topic = getTopicById(state.steps, topicId as number);
+
+        const previousField =
+          topic.fields[(dragDestination?.index as number) - 1];
+        const nextField = topic.fields[dragDestination?.index as number];
```

Comment:
I think we could check if `dragDestination` exists at the beginning then we can avoid the castings. That should make the compiler happy while making the code more safe. The same goes for the other variables used.

Reply:
I think I check this on line `107` but I should put the check on `dragDestination` instead of `result.destination` that will be better.